rfcs/0000-gateway-api-metrics-exporter.md

```diff
@@ -0,0 +1,197 @@
+        observedGeneration: 1
+        reason: Ready
+        status: "True"
+        type: Ready
+      - lastTransitionTime: "2023-08-15T13:22:06Z"
+        message: No errors found
+        observedGeneration: 1
+        reason: ResolvedRefs
+        status: "True"
+        type: ResolvedRefs
+      name: api
+```
+
+and HTTPRoute parents status conditions:
+
+```yaml
+  status:
+    parents:
+    - conditions:
+      - lastTransitionTime: "2024-05-16T16:17:38Z"
+        message: Object affected by AuthPolicy default/toystore
+        observedGeneration: 1
+        reason: Accepted
+        status: "True"
+        type: kuadrant.io/AuthPolicyAffected
+      - lastTransitionTime: "2024-05-16T16:18:51Z"
+        message: Object affected by RateLimitPolicy default/toystore
+        observedGeneration: 1
+        reason: Accepted
+        status: "True"
+        type: kuadrant.io/RateLimitPolicyAffected
+      controllerName: kuadrant.io/policy-controller
+      parentRef:
+        group: gateway.networking.k8s.io
+        kind: Gateway
+        name: api-gateway
+        namespace: kuadrant-system
+    - conditions:
+      - lastTransitionTime: "2024-05-16T16:17:38Z"
+        message: Object affected by AuthPolicy default/toystore
+        observedGeneration: 1
+        reason: Accepted
+        status: "True"
+        type: kuadrant.io/AuthPolicyAffected
+      - lastTransitionTime: "2024-05-16T16:18:51Z"
+        message: Object affected by RateLimitPolicy default/toystore
+        observedGeneration: 1
+        reason: Accepted
+        status: "True"
+        type: kuadrant.io/RateLimitPolicyAffected
+      - lastTransitionTime: "2024-05-20T11:45:33Z"
+        message: Route was valid
+        observedGeneration: 1
+        reason: Accepted
+        status: "True"
+        type: Accepted
+      - lastTransitionTime: "2024-05-20T11:45:33Z"
+        message: All references resolved
+        observedGeneration: 1
+        reason: ResolvedRefs
+        status: "True"
+        type: ResolvedRefs
+      controllerName: istio.io/gateway-controller
+      parentRef:
+        group: gateway.networking.k8s.io
+        kind: Gateway
+        name: api-gateway
+        namespace: kuadrant-system
+```
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+First, implement the existing Gateway API metrics that are part of the [gateway-api-state-metrics](https://github.com/Kuadrant/gateway-api-state-metrics/blob/main/METRICS.md) project.
+This will allow the new prometheus exporter to replace that project as a whole.
+The metrics will be backwards compatible with the gateway-api-state-metrics project, with the addition of new metrics.
+
+Second, implement new metrics, as per the examples below, to capture the additional status information:
+
+```promql
+gatewayapi_gateway_status_listeners_conditions{namespace="<NAMESPACE>",name="<GATEWAY>",listener_name="<LISTENER_NAME>",type="<ResolvedRefs|Ready|Other>} 1
```

Comment:
Wdyt about using a gateway API acronym so something like gwapi for the metric name? im thinking long run having to type these out might get tedious. 
```suggestion
gwapi_gateway_status_listeners_conditions{namespace="<NAMESPACE>",name="<GATEWAY>",listener_name="<LISTENER_NAME>",type="<ResolvedRefs|Ready|Other>} 1
```

Reply:
I do like the shorter version.
My concern with changing the name of the metrics is that they lose [backwards compatibility](https://github.com/Kuadrant/architecture/pull/101/files#diff-c3c9a29408a0e64d785783106086dd38d7db6e8aac81f282eae480fdc1820974R109) with gateway-api-state-metrics, and any example alerts and dashboards provided as part of that. Same for the kuadrant example alerts and dashboards.
We would need to allow for migration of versions.
Do you think it's worth the change?

Reply:
yes good point, i think it would be nice to have but the efforts to change it arent worth it